libdrgn/python/thread.c

```diff
@@ -56,6 +56,21 @@ static DrgnObject *Thread_get_object(Thread *self)
 	return_ptr(ret);
 }
 
+static PyObject *Thread_get_name(Thread *self)
+{
+	_cleanup_free_ char *ret = NULL;
```

Comment:
If we want to use `_cleanup_free_` for this (which I think we do), then we need `drgn_thread_name()` to guarantee that it only modifies `*ret` on success instead of saying that its contents are undefined on error. For context, before drgn made use of `_cleanup_` attributes, I usually made no such guarantees, but now I've tended to provide that guarantee (e.g., https://github.com/osandov/drgn/commit/e91fcbe22a2c5fcaf8b8765323023919499deb92).

That will just mean using a temporary in a few places instead of assigning directly to `*ret`. E.g., instead of

```c
*ret = strdup(...);
if (!*ret)
	return &drgn_enomem;
```

do this:

```c
char *tmp = strdup(...);
if (!tmp)
	return &drgn_enomem;
*ret = tmp;
```